### PR TITLE
Implement methods with `Any + Send + Sync` bounds

### PR DESCRIPTION
**Pull request by ldesgoui**
Hello,

Seems like a straightforward change, I don't believe I'm doing anything cursed

Thanks for the crate

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -66,10 +66,36 @@ pub type StaticTypeMap = GenericStaticTypeMap<dyn Any>;
 ///
 /// type_map.remove::<bool>();
 ///
-/// println!("Our `StaticTypeMap` contains {} types.", type_map.len());
+/// println!("Our `SendStaticTypeMap` contains {} types.", type_map.len());
 /// ```
 pub type SendStaticTypeMap = GenericStaticTypeMap<dyn Any + Send>;
 
+/// A wrapper over `HashMap<TypeId, Box<dyn Any + SendSync>>`.
```
Comment (malobre) [inline src/lib.rs]:
```suggestion
/// A wrapper over `HashMap<TypeId, Box<dyn Any + Send + Sync>>`.
```

File: src/lib.rs [hunk 2]
```diff
@@ -249,7 +275,7 @@ macro_rules! impl_bounded_map {
                 {
                     self.0
                         .get(&TypeId::of::<T>())
-                        .and_then(|any| any.downcast_ref())
+                        .and_then(|any| (&**any as &(dyn Any)).downcast_ref())
```
Comment (malobre) [inline src/lib.rs:278]: Why ?

**Review by malobre — CHANGES_REQUESTED** [review]

**Comment by malobre** [comment]
Thanks for the PR !

File: src/lib.rs (hunk 2 above)
Comment (ldesgoui) [reply src/lib.rs:278]: any is potentially a Box<dyn Any + Send + Sync> and downcast_ref is not available in that case

**Review by malobre — APPROVED** [review]

**Comment by malobre** [comment]
I'll publish a new version tomorrow.

**Pull request merged**